### PR TITLE
Blocked nav #135

### LINKED ISSUE #135: User Story 6, Users cannot navigate to certain paths
```
Users should see a 404 error under the following conditions:
- if visitors try to navigate to any /profile path
- if visitors try to navigate to any /dashboard path
- if visitors try to navigate to any /admin path
- if registered users try to navigate to any /dashboard path
- if registered users try to navigate to any /admin path
- if merchants try to navigate to any /profile path
- if merchants try to navigate to any /admin path
- if merchants try to navigate to any /cart path
- if admin users try to navigate to any /profile path
- if admin users try to navigate to any /dashboard path
- if admin users try to navigate to any /cart path

If you think of any additional pages to block, please do so.
```

### Checklist:

- [ ] base code tests are written
- [ ] base code is written to pass tests
- [ ] edge case ("sad path") tests written (if required)
- [ ] edge case code written (if required)
- [ ] code is reviewed
- [ ] branch is merged
- [ ] all tests still pass

depends on #139

depends on #138

depends on #137

### PR DESCRIPTION
**Pull request by stoic-plus**
closes #135 

* adds before action to `CartItemsController` - only visitor or defaults should be able to see
* removes skipping `not_today_satan` from `merchants#show` 
  (/dashboard from what Ian has said is only for merchants - admin should have dashboard under `admin/dashboard`) 
  * So conditional is not needed to check if merchant or admin because only merchants should see

* using `render_404` in more places (from app controller)
* adds helper method to `UsersController` to determine if current user and is default user

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -24,5 +24,4 @@ def empty_cart
   def render_404
```
Comment (tonyr729) [inline app/controllers/application_controller.rb]: Nice
